module/chunks/chunkVerifier_test.go

```diff
@@ -346,3 +396,34 @@ func (vm *vmMock) Run(ctx fvm.Context, proc fvm.Procedure, led state.View, progr
 
 	return nil
 }
+
+type vmSystemOkMock struct{}
+
+func (vm *vmSystemOkMock) Run(ctx fvm.Context, proc fvm.Procedure, led state.View, programs *programs.Programs) error {
+	tx, ok := proc.(*fvm.TransactionProcedure)
+	if !ok {
+		return fmt.Errorf("invokable is not a transaction")
+	}
+
+	tx.ServiceEvents = []flow.Event{epochSetupEvent}
+
+	_, _ = led.Get("00", "", "")
+	_, _ = led.Get("05", "", "")
+	_ = led.Set("05", "", "", []byte{'B'})
+	tx.Logs = []string{"log1", "log2"}
+
+	return nil
+}
+
+type vmSystemBadMock struct{}
+
+func (vm *vmSystemBadMock) Run(ctx fvm.Context, proc fvm.Procedure, led state.View, programs *programs.Programs) error {
+	tx, ok := proc.(*fvm.TransactionProcedure)
+	if !ok {
+		return fmt.Errorf("invokable is not a transaction")
+	}
+
+	tx.ServiceEvents = []flow.Event{epochCommitEvent}
```

Comment:
```suggestion
	// EpochSetup event is expected, but we emit EpochCommit here resulting in a chunk fault
	tx.ServiceEvents = []flow.Event{epochCommitEvent}
```
Same here, a comment noting why this mock is expected to fail